src/Raven.Server/Documents/Handlers/Processors/Indexes/IndexHandlerProcessorForConvertAutoIndex.cs

```diff
@@ -0,0 +1,716 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Formatting;
+using Raven.Client;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Indexes.Spatial;
+using Raven.Client.Exceptions.Documents.Indexes;
+using Raven.Server.Documents.Indexes;
+using Raven.Server.Json;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Indexes;
+
+internal class IndexHandlerProcessorForConvertAutoIndex<TRequestHandler, TOperationContext> : AbstractHandlerProcessor<TRequestHandler>
+    where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+    where TOperationContext : JsonOperationContext
+{
+    public IndexHandlerProcessorForConvertAutoIndex([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+    {
+    }
+
+    private string GetName() => RequestHandler.GetStringQueryString("name", required: true);
+
+    private ConvertType GetConvertType()
+    {
+        var typeAsString = RequestHandler.GetStringQueryString("type", required: true);
+
+        if (Enum.TryParse(typeAsString, ignoreCase: true, out ConvertType convertType) == false)
+            throw new InvalidOperationException($"Could not parse '{typeAsString}' to any known conversion type.");
+
```

Comment:
I wonder whether we should introduce `?language=` parameter. Currently we'll support only C# syntax and throw on JavaScript

Reply:
Let's leave it, but I would call it 'syntax' or sth like that